spec/request/profile_spec.rb

```diff
@@ -0,0 +1,84 @@
+require "rails_helper"
+
+describe "Profile", type: :request do
+  let!(:profile) { create(:profile) }
+  let(:other_user) { create(:user) }
+  let(:new_attributes) { attributes_for(:profile).stringify_keys! }
+  let(:expected_attributes) { profile.attributes.merge(new_attributes) }
+
+  describe "#update" do
+    context "with inappropriate user" do
+      before do
+        login_as(other_user, scope: :user)
+      end
+
+      it "should fail to update the profile" do
+        expect {
+          put "/profiles/#{profile.id}", params: { profile: new_attributes }
+        }.not_to change {
+          profile.reload.attributes
+        }
+
+        expect(response).to redirect_to(profiles_path)
+        expect(response).to have_http_status(:redirect)
+        expect(flash[:alert]).to eq("You do not own this profile!")
```

Comment:
You can generalize problem of trying to access/edit not permitted object.

you can change in controller:
```
def set_profile
  @profile = Profile.find(params[:id])
end
```
to 
```
def set_profile
  @profile = current_user.profiles.find(params[:id])
end
```
If a user tries to find not his profile then RecordNotFound will be raised so you don't need `check_owner!` anymore. As trying to access not mine data is not normal user action we don't need to take care of UX here. We can just render generic error or 404 page. You can use `rescue_from RecordNotFound` in `ApplicationController` to achieve that